contrib/endpoints/src/api_manager/service_management_fetch_test.cc

```diff
@@ -0,0 +1,171 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "contrib/endpoints/src/api_manager/service_management_fetch.h"
+
+#include "contrib/endpoints/src/api_manager/context/global_context.h"
+#include "contrib/endpoints/src/api_manager/context/request_context.h"
+#include "contrib/endpoints/src/api_manager/context/service_context.h"
+#include "contrib/endpoints/src/api_manager/mock_api_manager_environment.h"
+#include "contrib/endpoints/src/api_manager/mock_request.h"
+
+using ::testing::_;
+using ::testing::Invoke;
+using ::testing::Mock;
+using ::testing::Return;
+
+using ::google::api_manager::utils::Status;
+
+namespace google {
+namespace api_manager {
+namespace {
+
+const char kServiceManagementService[] =
+    "https://servicemanagement.googleapis.com";
+const char kServiceManagementServiceManager[] =
+    "/google.api.servicemanagement.v1.ServiceManager";
+
+const char kServiceConfig[] = R"(
+{
+  "name": "endpoints-test.cloudendpointsapis.com",
+  "control": {
+     "environment": "http://127.0.0.1:808"
+  }
+})";
+
+const char kServiceConfig1[] = R"(
+{
+  "name": "bookstore.test.appspot.com",
+  "title": "Bookstore",
+  "id": "2017-05-01r1"
+}
+)";
+
+const char kServerConfigWithServiceNameConfigId[] = R"(
+{
+  "google_authentication_secret": "{}",
+  "metadata_server_config": {
+    "enabled": true,
+    "url": "http://localhost"
+  },
+  "service_control_config": {
+    "report_aggregator_config": {
+      "cache_entries": 10000,
+      "flush_interval_ms": 1000001232
+    },
+    "quota_aggregator_config": {
+      "cache_entries": 300000,
+      "refresh_interval_ms": 1000
+    }
+  },
+  "service_name": "service_name_from_server_config",
+  "config_id": "2017-05-01r1"
+}
+)";
+
+}  // namespace
+
+class service_management_fetchTest : public ::testing::Test {
+ public:
+  void SetUp() {
+    env_.reset(new ::testing::NiceMock<MockApiManagerEnvironment>());
+    // save the raw pointer of env before calling std::move(env).
+    raw_env_ = env_.get();
+
+    global_context_ = std::make_shared<context::GlobalContext>(
+        std::move(env_), kServerConfigWithServiceNameConfigId);
+
+    if (global_context_->service_account_token()) {
+      // register auth token for servicemanagement services
+      global_context_->service_account_token()->SetAudience(
+          auth::ServiceAccountToken::JWT_TOKEN_FOR_SERVICEMANAGEMENT_SERVICES,
+          std::string(kServiceManagementService) +
+              kServiceManagementServiceManager);
+    }
+
+    std::unique_ptr<MockRequest> request(
+        new ::testing::NiceMock<MockRequest>());
+
+    std::unique_ptr<Config> config = Config::Create(raw_env_, kServiceConfig);
+    ASSERT_NE(config.get(), nullptr);
+
+    global_context_->server_config()
```

Comment:
Still trying to find the root cause the link failure when I removed "Config::Create(raw_env_, kServiceConfig);"